bitcoin/src/sign_message.rs

```diff
@@ -126,6 +125,15 @@ mod message_signing {
             })
         }
 
+        /// Create from a byte slice.
+        #[deprecated(since = "0.32.0", note = "Use `from_byte_array` instead.")]
+        pub fn from_slice(bytes: &[u8]) -> Result<MessageSignature, MessageSignatureError> {
+            // Convert the Vec<u8> into a fixed-length array
+            let byte_array: [u8; 65] =
+                    bytes.try_into().map_err(|_| MessageSignatureError::InvalidLength)?;
+            MessageSignature::from_byte_array(&byte_array)
+        }
```

Comment:
Perhaps this is cleaner?

```rust
        pub fn from_slice(bytes: &[u8]) -> Result<MessageSignature, MessageSignatureError> {
            Self::from_byte_array(
                &<[u8; 65]>::try_from(bytes).map_err(|_| MessageSignatureError::InvalidLength)?,
            )
        }
```

Reply:
I prefer the former, though it should split up the methods. We might have misconfigured rustfmt.

Reply:
Due to one vote each way I have left it as it is